src/org/mozilla/javascript/NativeArray.java

```diff
@@ -862,17 +862,15 @@ static long getLengthProperty(Context cx, Scriptable obj, boolean throwIfTooLarg
         }
 
         double doubleLen = ScriptRuntime.toNumber(len);
+
+        // ToLength
         if (doubleLen > NativeNumber.MAX_SAFE_INTEGER) {
-            if (throwIfTooLarge) {
```

Comment:
It seems that this is probably the policy.
- get length
  - Up to 53 ** 2 - 1
  - https://262.ecma-international.org/11.0/#sec-tolength
- set length
  - Up to 31 ** 2 - 1
  - https://262.ecma-international.org/11.0/#sec-arraycreate